UI/src/main/java/adList/Ad.java

```diff
@@ -0,0 +1,157 @@
+package adList;
+
+import com.google.gson.Gson;
+
+import java.util.*;
+
+public class Ad {
+    private final String id;
+    private String description;
+    private final Date createdAt;
+    private String link;
+    private String vendor;
+    private String photoLink;
+    private List<String> hashTags;
+    private String discount;
+    private Date validUntil;
+    private Double rating;
+    private List<String> reviews;
+
+    public Ad(String id, String description, Date createdAt, String link, String vendor, List<String> hashTags, String discount, Date validUntil) {
+        this.id = id;
+        this.description = description;
+        this.createdAt = createdAt != null ? (Date) createdAt.clone() : null;
+        this.link = link;
+        this.vendor = vendor;
+        this.hashTags = hashTags != null ? new ArrayList<>(hashTags) : null;
+        this.discount = discount;
+        this.setValidUntil(validUntil);
+    }
+    public Ad(Ad ad){
+        if(ad != null) {
+            this.setHashTags(ad.getHashTags());
+            this.id = ad.getId();
+            this.setDescription(ad.getDescription());
+            this.createdAt = ad.getCreatedAt();
+            this.setLink(ad.getLink());
+            this.setVendor(ad.getVendor());
+            this.setPhotoLink(ad.getPhotoLink());
+            this.setHashTags(ad.getHashTags());
+            this.setDiscount(ad.getDiscount());
+            this.setValidUntil(ad.getValidUntil());
+            this.setRating(ad.getRating());
+            this.setRating(ad.getRating());
+        }
+        else {
+            id = null;
+            createdAt = null;
+        }
+    }
+    public Ad(String json){
+        this(new Gson().fromJson(json, Ad.class));
+    }
+    public String getDescription() {
+        return description;
+    }
+
+    public boolean setDescription(String description) {
+        if(description !=null && description.length() < 200) {
+            this.description = description;
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public String getLink() {
+        return link;
+    }
+
+    public void setLink(String link) {
+        this.link = link;
+    }
+
+    public String getVendor() {
+        return vendor;
+    }
+
+    public void setVendor(String vendor) {
+        this.vendor = vendor.trim().equals("")? "vendor" + id : vendor;
```

Comment:
That's OK. You can also use `StringUtils.isBlank()` from Apache Commons (google it). Apache Commons is one of the most used Java packages in all